### PR TITLE
Migrate remaining files to TS

### PR DESCRIPTION
**Pull request by blazejkustra**
### Details
This PR aims to migrate remaining files to TS (other than Onyx.js and withOnyx.js). Changes are almost only on the type level, so it shouldn't create any regressions.

### Related Issues

https://github.com/Expensify/App/issues/34342

### CODE REVIEW

**Review by BartoszGrajdek — APPROVED** [review]

File: lib/createDeferredTask.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
+type DeferredTask = {
+    promise: Promise<unknown>;
+    resolve: (value: unknown) => void;
+};
```
Comment (fabioh8010) [inline lib/createDeferredTask.ts:4]:
```suggestion
type DeferredTask = {
    promise: Promise<unknown>;
    resolve?: (value: unknown) => void;
};
```

File: lib/SyncQueue.js
Comment (fabioh8010) [inline lib/SyncQueue.js:1]: Is this file not being used anymore?

File: lib/createDeferredTask.ts (hunk 1 above)
Comment (fabioh8010) [reply lib/createDeferredTask.ts:4]: `resolve` is initially undefined until the promise is resolved.

File: lib/createDeferredTask.ts [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
  * The returned promise will complete when we call `resolve`
  * Useful when we want to wait for a tasks that is resolved from an external action
- *
- * @template T
- * @returns {{ resolve: function(*), promise: Promise<T|void> }}
  */
-export default function createDeferredTask() {
-    const deferred = {};
+export default function createDeferredTask(): DeferredTask {
+    const deferred: DeferredTask = {} as DeferredTask;
```
Comment (fabioh8010) [inline lib/createDeferredTask.ts:12]:
```suggestion
    const deferred = {} as DeferredTask;
```

wdyt?

File: lib/createDeferredTask.ts (hunk 1 above)
Comment (fabioh8010) [reply lib/createDeferredTask.ts:4]: I updated my code suggestion, it should be `void`.

File: lib/SyncQueue.js
Comment (blazejkustra) [reply lib/SyncQueue.js:1]: Yes 👍

**Review by fabioh8010 — APPROVED** [review]

**Review by arosiclair — APPROVED** [review]

**Pull request merged**